reactioncommerce/reaction
# Demeteorize and reorg some core startup code

Resolves #5210 
Resolves #5219
Impact: **minor**  
Type: **feature**

Part of epic https://github.com/reactioncommerce/reaction/issues/4996

## Changes
- Removed `Reaction.init` function and moved that code to where it was called in the main startup function
- Removed support for undocumented feature `Meteor.settings.cdnPrefix`
- Removed support for undocumented feature `Reaction.Endpoints`
- Removed support for prerender by `prerender-node` NPM package
- Updated `handlebars` package to 4.1.2 to fix snyk warning
- Moved job server startup to `imports/utils/jobs.js` (still Meteor)
- Moved accounts/groups startup code to `accounts` core plugin and de-Meteorize
- Update and de-Meteorize sample data import code. On an empty database at startup, a primary shop, a sample product, and related sample tags are still inserted as before. The `Shipping` collection no longer has sample shipping methods added. The data itself is moved from `.json` files in `/private/data` (a Meteor thing) to a single JS file `/imports/plugins/core/core/server/no-meteor/util/sampleData.js`
- Reorg the execution order of startup code based on dependency needs. Run migrations later to be sure everything they need is there.
- Partially de-Meteorize import into the `Translations` collection on startup. Some translation files are still loaded in Meteor into the `Assets` collection, but moving from `Assets` to per-shop documents in `Translations` is now done without Meteor, in an "afterShopCreate" listener.
- Set log level to "ERROR" when running Jest tests to keep the output clean

## Breaking changes
- Removed support for undocumented feature `Meteor.settings.cdnPrefix`
- Removed support for undocumented feature `Reaction.Endpoints`
- Removed support for prerender by `prerender-node` NPM package

## Testing
1. Start the app with a fresh database. Verify no errors in the startup logging.
2. Verify sample data was inserted.
3. Verify that Groups and user roles are inserted/updated for the shop as before.
4. Smoke test the operator UI
5. Configure emailing and verify that emails are sent (because job server code was moved)
6. Also try starting up against a non-empty database, such as a sample data set. Verify no errors in startup logging.
